### PR TITLE
Advanced ToDo Example

### PR DESCRIPTION
**Pull request by austin94**
Hi @KevinGrandon. In https://github.com/fusionjs/fusionjs.github.io/issues/18 you mentioned that this boilerplate repo might be a good place to try out a more advanced example so I threw together a quick Todo example.

I tried to combine some _fusion specific_ things into this and was wondering if you had any feedback and/or if you think this might be something you would be useful as an example boilerplate?

## Fusion-Specific Features

- Create a custom fusion plugin to include [todomvc-app-css](https://www.npmjs.com/package/todomvc-app-css) styling for the application
- Use `fusion-plugin-react-redux` and `fusion-plugin-react-router` with a standard action/container/reducer pattern for all of the business logic
- Use `fusion-plugin-i18n-react` to provide translations for all content
  - Provide English, German, French, and Japanese translations by default
  - Allow users to toggle language without changing their browsers language settings
- Provide a custom `I18nLoader` to allow locale based routing
  - Route breakdown:
    - `/en`, `/fr`, etc. - Include translations from "best match" locale. Will redirect if not a direct match.
    - `/` - Use the requests `accept-language` header to determine which locale route to redirect to.

## Translation Examples

en

de

ja

## While I Was Here

There was a stray `yarn-error.log` file in one of the other examples. I updated the `.gitignore` to match the other examples and remove it.

## Not Finished

- Need to add test coverage
- Normally I like to include at an right-to-left (RTL) translation example. From a quick look through it seems that the i18n plugin has no concept of RTL locales. In my day-to-day work, knowing if a current translation if RTL or LTR is almost as important as the actual translation. Do you know if this is something that is within scope for that package?
- The custom `i18nLoader` plugin I wrote for the dynamic routing is broken ☹️ Despite being behind a `__NODE__` check it seems that something server dependent is getting triggered if the imports are uncommented.
- Create a more detailed README with implementation details if this is something you all are interested in.

### CODE REVIEW

**Comment by KevinGrandon** [comment]
Hi @austin94 - Apologies for missing this in my Github notifications. It's really awesome that you dug into Fusion.js and put this together! 

I'm happy to land this here, but was wondering if you would want to open source this under a repo under your user? It's possible that this gets moved to another org in the future, but if you did put it under your name you would get any current open source glory, and the joys of maintaining as well. (It's completely up to you though).

File: todo-boilerplate/config/config.js.tpl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export default const config = {
```
Comment (KevinGrandon) [inline todo-boilerplate/config/config.js.tpl]: nit: probably don't need this file.

File: todo-boilerplate/src/plugins/TodoStylingPlugin.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { assetUrl, dangerouslySetHTML, createPlugin } from 'fusion-core';
+
+export default createPlugin({
```
Comment (KevinGrandon) [inline todo-boilerplate/src/plugins/TodoStylingPlugin.js]: Should probably be updated to use our new [helmet plugin](https://github.com/fusionjs/fusion-plugin-react-helmet-async#usage)

**Comment by austin94** [comment]
> but if you did put it under your name you would get any current open source glory

Haha it's all about the glory. I'll go ahead and plan on moving this over to https://github.com/austin94/fusion-todo-boilerplate soon

It looks like there have been some additions since I last looked ([create-fusion-app](https://www.npmjs.com/package/create-fusion-app), helment plugin, etc.). Once I address the issues above I'll probably start from a clean slate in that repo.

Thanks for taking a look!

**Comment by austin94** [comment]
@KevinGrandon I've moved this code (with the suggested changes to) https://github.com/austin94/fusion-todo-boilerplate

I was able to clean it up quite a bit and fix the `I18nLoader` issues that I mentioned above. Do you think this is something that would be of interest to others? If so I can open a PR to add this to the examples list for the documentation site.

Thanks again for the help with this!

**Comment by KevinGrandon** [comment]
That's awesome, thanks so much. We would definitely include a link to your example in the docs. Going to close this PR now that you have moved the code.

Regarding the i18nLoader issue, I'm not sure what's going on there. We might be able to add [this option](https://fusionjs.com/api/fusion-cli/docs/fusionrc#assumenoimportsideeffects) to a .fusionjsrc.js file at the root of the project.

**Pull request closed without merging**